docs/features/software-templates/writing-templates.md

```diff
@@ -46,6 +46,18 @@ spec:
           ui:options:
             catalogFilter:
               kind: Group
+        user:
+          title: User
+          type: string
+          description: Owner of the component
+          ui:field: EntityPicker
+          ui:options:
+            allowArbitraryValues: false
+            nameTemplate: ${{entity.spec.profile.displayName}} (${{entity.spec.profile.email}})
```

Comment:
Really torn on introducing this tbh. Using the same templating syntax here as in the steps I feel has a very high risk of causing confusion. It's also a bespoke implementation although arguably we can share it if needed.

Another direction here is to simply say that this level of customization is not something that the `EntityPicker` supports in YAML. To do this you'd instead need to wrap up the `EntityPicker` implementation in code an expose that as a separate field extension, like the [`OwnerPicker`](https://github.com/backstage/backstage/blob/ba3b7d1782ee4acdb0a391ecf88176333f9b3029/plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx#L28) does. I'm leaning quite heavily towards that to be honest, being able to customize the string formatting in code seems desirable here. You might pretty quickly for example want to be able to provide fallback values or apply additional transforms to the text here.

Thoughts on this @benjdlambert?

Reply:
Hmm, I'm also feeling that there's a small forwards compatibility here that we're looking to see what it would mean to introduce the `${{ }}` syntax in the `parameters` block to be able to do some of this templating over here: #16275.

I'm feeling that adding this functionality that we want to pass the raw string through instead of templating it causes a lot of confusion here.

Kind of leaning towards what you're suggesting @Rugvip by wrapping up this component in a separate FieldExtension might be the best choice to move this customisation out of the yaml layer makes sense.